File: examples/kotlin/TestUsage.kt
import uniffi.smpp_codec_ffi.*

fun main() {
    println("Testing Kotlin Bindings...")

    try {
        // Create SubmitSmRequest
        val request = SubmitSmRequest(
            sequenceNumber = 1u,
            serviceType = "",
            sourceAddrTon = Ton.UNKNOWN,
            sourceAddrNpi = Npi.UNKNOWN,
            sourceAddr = "123456",
            destAddrTon = Ton.INTERNATIONAL,
            destAddrNpi = Npi.ISDN,
            destinationAddr = "9876543210",
            esmClass = 0u,
            protocolId = 0u,
            priorityFlag = 0u,
            scheduleDeliveryTime = null,
            validityPeriod = null,
            registeredDelivery = 1u,
            replaceIfPresentFlag = 0u,
            dataCoding = 0u,
            smDefaultMsgId = 0u,
            shortMessage = "Hello, World!".encodeToByteArray().toList(),
            tlvs = emptyList()
        )

        println("Created Request: $request")

        // Encode
        val encoded = encodeSubmitSmRequest(request)
        println("Encoded bytes: $encoded")

        // Decode
        val decoded = decodeSubmitSmRequest(encoded)
        println("Decoded Request: $decoded")

        // Simple Assertion
        if (request.sequenceNumber == decoded.sequenceNumber && 
            request.sourceAddr == decoded.sourceAddr) {
            println("SubmitSm Test Passed!")
        } else {
            println("SubmitSm Test FAILED!")
        }

    } catch (e: Exception) {
        println("Error: $e")
    }
}

File: examples/swift/TestUsage.swift
import Foundation
import smpp_codec_ffi

func testSubmitSm() {
    print("Testing Swift Bindings (SubmitSm)...")

    let request = SubmitSmRequest(
        sequenceNumber: 1,
        serviceType: "",
        sourceAddrTon: .unknown,
        sourceAddrNpi: .unknown,
        sourceAddr: "123456",
        destAddrTon: .international,
        destAddrNpi: .isdn,
        destinationAddr: "9876543210",
        esmClass: 0,
        protocolId: 0,
        priorityFlag: 0,
        scheduleDeliveryTime: nil,
        validityPeriod: nil,
        registeredDelivery: 1,
        replaceIfPresentFlag: 0,
        dataCoding: 0,
        smDefaultMsgId: 0,
        shortMessage: Data("Hello, World!".utf8),
        tlvs: []
    )

    print("Created Request: \(request)")

    // Encode
    let encoded = encodeSubmitSmRequest(request: request)
    print("Encoded bytes: \(encoded)")

    // Decode
    do {
        let decoded = try decodeSubmitSmRequest(buffer: encoded)
        print("Decoded Request: \(decoded)")

        if request.sequenceNumber == decoded.sequenceNumber && 
           request.sourceAddr == decoded.sourceAddr {
            print("SubmitSm Test Passed!")
        } else {
            print("SubmitSm Test FAILED!")
        }
    } catch {
        print("Decoding failed: \(error)")
    }
}

testSubmitSm()
